pkg/shim/scheduler.go

```diff
@@ -213,7 +213,7 @@ func (ss *KubernetesShim) recoverSchedulerState(e *fsm.Event) {
 				recoverableAppManagers = append(recoverableAppManagers, m)
 			}
 		}
-		if err := ss.context.WaitForRecovery(recoverableAppManagers, 30*time.Second); err != nil {
+		if err := ss.context.WaitForRecovery(recoverableAppManagers, 5*time.Minute); err != nil {
```

Comment:
Increased this to avoid potential issues on large clusters. However, the only way a node should fail to transition is if the scheduler is very badly misconfigured.

Reply:
Around mid 2021, I used to observe that recovery could fail relatively frequently. When it did, it never recovered and we could benefit from knowing the failure sooner rather than later. That's why @chenya-zhang introduced this change to fail really quickly (in 30 secs). A ton of things have changed over the last year, 5 minutes should be fine. But I'll keep an eye on what if any failed recovery scenarios could occur

Reply:
I also debated what to use here for a timeout. I went through the code paths that the recover event goes through, and it seems like the only error that can occur is if the RM is not properly registered with the shim (which should never happen). If it were to happen, it would fail regardless of how long a timeout we provide.

My reasoning for a longer timeout is that while unlikely, it's not impossible that we fail to finish recovery with a large enough cluster. Also, we log every second how many nodes are waiting on recovery, so if recovery fails we should be able to see this in the logs pretty clearly.

Personally, I'd rather not have a timeout at all and just do the recovery synchronously, but that's a more invasive change at this point...